test/e2e/mergeRequiredFromAllOf.ts

```diff
@@ -0,0 +1,31 @@
+export const input = {
+  title: 'MergeRequiredFromAllOf',
+  type: 'object',
+  properties: {
+    reference: {
```

Comment:
What happens when there's a second property that also references `#/definitions/defined`, but doesn't have the `required`?